Sources/ArcGISToolkit/Components/Bookmarks/Bookmarks.swift

```diff
@@ -130,47 +100,128 @@ public struct Bookmarks: View {
     ) {
         self.geoModel = geoModel
         self.viewpoint = viewpoint
+        self.bookmarks = []
         _isPresented = isPresented
     }
     
     public var body: some View {
-        VStack {
-            BookmarksHeader(isPresented: $isPresented)
-                .padding([.horizontal, .top])
-                .onSizeChange {
-                    headerHeight = $0.height
-                }
-            ScrollView {
-                VStack {
-                    if geoModel == nil || isGeoModelLoaded {
-                        BookmarksList(bookmarks: bookmarks)
-                            .onSelectionChanged {
-                                selectBookmark($0)
-                            }
-                    } else {
-                        loadingView
+        BookmarksHeader(isPresented: $isPresented)
+            .padding([.horizontal, .top])
+        Divider()
+        if !bookmarks.isEmpty {
+            list
+                .onChange(of: selectedBookmark) { selectedBookmark in
+                    if let selectedBookmark {
+                        selectBookmark(selectedBookmark)
                     }
                 }
-                .onSizeChange {
-                    listHeight = $0.height
+        } else if let loadingError {
+            makeErrorMessage(with: loadingError)
+        } else if geoModel != nil {
```

Comment:
If there is a `geoModel`, but it does not have any bookmarks, it will always show the `loading` view. It will never fall through to the `noBookmarks` view. Maybe something like:
```suggestion
        } else if geoModel != nil, !geoModel.isLoadingComplete {
```

```
extension GeoModel {
    var isLoadingComplete: Bool {
        loadStatus == .loaded || loadStatus == .failed
    }
}
```